src/Microsoft.Graph.Core/Serialization/DerivedTypeConverter.cs

```diff
@@ -38,17 +38,11 @@ public override bool CanConvert(Type objectType)
         /// <returns></returns>
         public override T Read(ref Utf8JsonReader reader, Type objectType, JsonSerializerOptions options)
         {
-            JsonDocument jsonDocument = JsonDocument.ParseValue(ref reader);
+            using JsonDocument jsonDocument = JsonDocument.ParseValue(ref reader);
             JsonElement type;
-            try
-            {
-                // try to get the @odata.type property if we can
-                if (!jsonDocument.RootElement.TryGetProperty(CoreConstants.Serialization.ODataType, out type))
-                {
-                    type = default;
-                }
-            }
-            catch (InvalidOperationException)
```

Comment:
Eliminates the try/catch from the hot path as we can check the `ValueKind` to avoid the `InvalidOperationException`